### PR TITLE
feat: Handles response with rate limit and with other errors

### PR DESCRIPTION
**Pull request by MatousJobanek**
Check if the rate limit has exceeded - if yes it tries to use the cached version and log warning, otherwise throws an exception with explaining message

### CODE REVIEW

File: drone-webdriver/src/main/java/org/jboss/arquillian/drone/webdriver/binary/downloading/source/GitHubSource.java [hunk 1]
```diff
@@ -175,4 +222,14 @@ protected String getUniqueKey() {
     protected GitHubLastUpdateCache getCache() {
         return cache;
     }
+
+    // workaround for https://github.com/SpectoLabs/hoverfly/issues/573
+    protected String getRateLimitRemainingHeaderKey() {
```
Comment (bartoszmajsak) [inline drone-webdriver/src/main/java/org/jboss/arquillian/drone/webdriver/binary/downloading/source/GitHubSource.java]: Is it really necessary to have workarounds in the production code for the tools we use in testing?

Comment (bartoszmajsak) [inline drone-webdriver/src/main/java/org/jboss/arquillian/drone/webdriver/binary/downloading/source/GitHubSource.java]: Does the issue still persist with `0.6.1`? https://github.com/SpectoLabs/hoverfly/issues/573#issuecomment-306159723

Comment (benjih) [inline drone-webdriver/src/main/java/org/jboss/arquillian/drone/webdriver/binary/downloading/source/GitHubSource.java]: Hey @bartoszmajsak,

0.6.1 of hoverfly-java should have this issue fixed.

Comment (MatousJobanek) [inline drone-webdriver/src/main/java/org/jboss/arquillian/drone/webdriver/binary/downloading/source/GitHubSource.java]: Yes, this is fixed in the release - haven't had time to upgrade it so far... Thank you @benjih

**Pull request merged**